lib/broadway/producer.ex

```diff
@@ -179,6 +207,17 @@ defmodule Broadway.Producer do
     {:noreply, [], state}
   end
 
+  # If we're here, we're sure that rate limiting was set up so we don't need to
+  # check if it's enabled.
+  def handle_info({__MODULE__, :reset_rate_limiting, allowed_count}, state) do
+    state = %{state | rate_limiting_allowed: allowed_count}
+
+    {messages_to_send, state} =
+      get_and_update_in(state.rate_limiting_buffer, &Enum.split(&1, allowed_count))
```

Comment:
This call should still go through the `maybe_rate_limit_and_buffer_messages`, otherwise, if the rate limit is 10, then we queue 20 while waiting for it to expire, we will emit 20 suddenly, going over the limit.

Reply:
We should probably write a test for this scenario.